### PR TITLE
Added the ability to configure custom pageViewHit (may useful for web analytics)

### PR DESCRIPTION
**Pull request by azhavoro**
Added the ability to configure custom pageViewHit (may useful for web analytics)

### Motivation and context
For SPA application it's important have ability to call virtual page view hit handler when new view is displayed.

### How has this been tested?
Manual testing

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat-ui/src/components/annotation-page/annotation-page.tsx [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 import './styles.scss';
 import React, { useEffect } from 'react';
+import { useLocation } from "react-router-dom";
```
Comment (bsekachev) [inline cvat-ui/src/components/annotation-page/annotation-page.tsx:7]: Please, use single quotes

File: cvat-ui/src/components/annotation-page/annotation-page.tsx [hunk 2]
```diff
@@ -13,6 +14,7 @@ import AnnotationTopBarContainer from 'containers/annotation-page/top-bar/top-ba
 import StatisticsModalContainer from 'containers/annotation-page/top-bar/statistics-modal';
 import StandardWorkspaceComponent from './standard-workspace/standard-workspace';
 import AttributeAnnotationWorkspace from './attribute-annotation-workspace/attribute-annotation-workspace';
+import { customWaViewHit } from 'utils/enviroment';
```
Comment (bsekachev) [inline cvat-ui/src/components/annotation-page/annotation-page.tsx:17]: Please, put it before relative import
Doesn't eslint show the error?

File: cvat-ui/src/components/create-model-page/create-model-page.tsx [hunk 3]
```diff
@@ -3,12 +3,14 @@
...
 import './styles.scss';
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useLocation } from "react-router-dom";
 import { Row, Col } from 'antd/lib/grid';
 import Text from 'antd/lib/typography/Text';
 
 import { ModelFiles } from 'reducers/interfaces';
 import CreateModelContent from './create-model-content';
+import { customWaViewHit } from 'utils/enviroment';
```
Comment (bsekachev) [inline cvat-ui/src/components/create-model-page/create-model-page.tsx:14]: The same like in the file above

File: cvat-ui/src/components/create-task-page/create-task-page.tsx [hunk 4]
```diff
@@ -4,13 +4,15 @@
...
 import React, { useEffect } from 'react';
+import { useLocation } from "react-router-dom";
 import { Row, Col } from 'antd/lib/grid';
 import Modal from 'antd/lib/modal';
 import Text from 'antd/lib/typography/Text';
 import Paragraph from 'antd/lib/typography/Paragraph';
 import TextArea from 'antd/lib/input/TextArea';
 
 import CreateTaskContent, { CreateTaskData } from './create-task-content';
+import { customWaViewHit } from 'utils/enviroment';
```
Comment (bsekachev) [inline cvat-ui/src/components/create-task-page/create-task-page.tsx:15]: The same like in the file above

File: cvat-ui/src/components/task-page/task-page.tsx [hunk 5]
```diff
@@ -15,6 +15,7 @@ import JobListContainer from 'containers/task-page/job-list';
 import ModelRunnerModalContainer from 'containers/model-runner-dialog/model-runner-dialog';
 import { Task } from 'reducers/interfaces';
 import TopBarComponent from './top-bar';
+import { customWaViewHit } from 'utils/enviroment';
```
Comment (bsekachev) [inline cvat-ui/src/components/task-page/task-page.tsx:18]: Please, put it before relative import

**Comment by nmanovic** [comment]
@azhavoro , need to resolve the conflict.

**Comment by nmanovic** [comment]
@azhavoro , will the approach work in our case? https://support.mouseflow.com/support/solutions/articles/44001535025-tracking-url-changes-with-react

**Comment by azhavoro** [comment]
> @azhavoro , will the approach work in our case? https://support.mouseflow.com/support/solutions/articles/44001535025-tracking-url-changes-with-react

It's the same approach based on Hooks and described only for React functional components. We use hooks for functional components and ComponentDidMount event for class component.

**Pull request merged**